### ISSUE TITLE
Invoke-IcingaCheckService has problems with spaces in servicename

### ISSUE THREAD

**Issue by stevie-sy** [Icinga/icinga-powershell-plugins#125]
Unfortunately, there are software companies which deliver setup routines that create a service with spaces.
e.g.
![image](https://user-images.githubusercontent.com/38723488/105348399-924d1e80-5be8-11eb-9eeb-5c400960d230.png)

The check "Invoke-IcingaCheckService" can't deal with this, because this is the plugin output:
![image](https://user-images.githubusercontent.com/38723488/105348488-b3157400-5be8-11eb-9954-40ae6178c973.png)

The director is rendering this call:
```
'C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe' '-C' 'try { Use-Icinga; } catch { Write-Output '\''The Icinga PowerShell Framework is either not installed on the system or not configured properly. Please check https://icinga.com/docs/windows for further details'\''; exit 3; }; Exit-IcingaPluginNotInstalled '\''Invoke-IcingaCheckService'\''; exit Invoke-IcingaCheckService' '-Service' 'EJM Agent' '-Exclude' '$null' '-Status' 'Running' '-Verbosity' '3'
```

I looked into the command definition and tested a little bit. I saw the quotes are missing. I'm not an Icinga-DSL pro, but this change works (for us). But I think there is a nicer and more elegant solution :
```
var arr = macro("$IcingaCheckService_Array_Service$");
if (len(arr) == 0) {
return "$null";
}
return "'" + arr.join("', '") + "'";
```

Tested with one service to check
![image](https://user-images.githubusercontent.com/38723488/105349471-089e5080-5bea-11eb-98a4-f0002396e742.png)
```
'C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe' '-C' 'try { Use-Icinga; } catch { Write-Output '\''The Icinga PowerShell Framework is either not installed on the system or not configured properly. Please check https://icinga.com/docs/windows for further details'\''; exit 3; }; Exit-IcingaPluginNotInstalled '\''Invoke-IcingaCheckService'\''; exit Invoke-IcingaCheckService' '-Service' ''\''"EJM Agent"'\''' '-Exclude' '$null' '-Status' 'Running' '-Verbosity' '3'
```

And also tested with more services to check:
![image](https://user-images.githubusercontent.com/38723488/105349546-279ce280-5bea-11eb-8a40-a9cd895df837.png)
```
'C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe' '-C' 'try { Use-Icinga; } catch { Write-Output '\''The Icinga PowerShell Framework is either not installed on the system or not configured properly. Please check https://icinga.com/docs/windows for further details'\''; exit 3; }; Exit-IcingaPluginNotInstalled '\''Invoke-IcingaCheckService'\''; exit Invoke-IcingaCheckService' '-Service' ''\''EJM Agent'\'', '\''icinga2'\'', '\''icingapowershell'\'', '\''MpsSvc'\'', '\''EventLog'\''' '-Exclude' '$null' '-Status' 'Running' '-Verbosity' '3'
```

**Comment by LordHepipud** [comment]
Hello and thank you for the issue.

There is a KB article available for Performance Counter: https://icinga.com/docs/icinga-for-windows/latest/doc/knowledgebase/IWKB000002/

There is no "proper" fix available, but simply to wrap the services with spaces in single quotes:

```
'EJM Agent'
```

on the service definition. This should resolve the issue.
If this proved to be correct, I would add another knowledge base article for this specifically.

**Comment by stevie-sy** [comment]
Yes, after I changed the argument 
![image](https://user-images.githubusercontent.com/38723488/105354322-b01e8180-5bf0-11eb-88a8-0f21e4ecd0d4.png)
from this 

`return arr.join(",");`

to this

`return "'" + arr.join("', '") + "'";`

it worked for us. This would be a possible fix for me. But as I wrote I'm not a Icinga-DSL pro.

**Comment by stevie-sy** [comment]
Additional info:
Whitout quotes we had also problems with MSSQL ressources (with $ in the name) defined as windows services 
e.g. MSSQL$SQLEXPRESS
![image](https://user-images.githubusercontent.com/38723488/105354964-9f224000-5bf1-11eb-87c4-a7ebd201c2b8.png)

with the change from above, also this is working fine now:
![image](https://user-images.githubusercontent.com/38723488/105355016-b82af100-5bf1-11eb-91df-bc30be3e3925.png)

**Comment by LordHepipud** [comment]
Yes indeed this will work and fix this issue, however, if you are going to use numeric depending numbers you will render a integer 0 to a string 0 for example.
I'm not sure if this actually will cause side effects and that's why I was not forcing a string rendering in this matter.

I will try to work a little more on this - maybe we can provide a more adaptive fix by parsing each single value and testing for whitespaces and wrapping them in strings then.

**Comment by stevie-sy** [comment]
Thank you, sounds good
